libs/api/protocol/src/lib/core/command/helper/set-base-data-from-command-on.entity.ts

```diff
@@ -0,0 +1,20 @@
+import { UserProducer } from '../../entity/partials/producer-partial.entity';
+import { ProtocolCommunicationEntryBase } from '../../entity/protocol-entries/protocol-entry-base.entity';
+import { BaseCreateMessageCommand } from '../base-create-message.command';
+
+export function setBaseDataFromCommandOnEntity(
+	cmd: BaseCreateMessageCommand,
+	entity: ProtocolCommunicationEntryBase,
+): void {
+	const producer = new UserProducer();
+	producer.userId = cmd.requestUser.id;
+	producer.firstName = cmd.requestUser.firstName;
+	producer.lastName = cmd.requestUser.lastName;
+
+	entity.time = cmd.time;
+	entity.sender = cmd.sender;
+	entity.recipient = cmd.recipient;
```

Comment:
`recipient` and `channel` only exist on communication messages. Other messages (e.g., status change) don't provide these fields. So either this method is for message-based entries only or has to check for the existence of those two fields.

Reply:
renamed to clarify